apps/application-system/api/src/app/modules/application/upload.processor.ts

```diff
@@ -1,44 +1,58 @@
 import { OnQueueCompleted, Process, Processor } from '@nestjs/bull'
 import { Job } from 'bull'
-// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
-// @ts-ignore
-import * as AmazonS3URI from 'amazon-s3-uri'
 import { ApplicationService } from './application.service'
 import { FileStorageService } from '@island.is/file-storage'
+import { Inject } from '@nestjs/common'
+import { ConfigType } from '@nestjs/config'
+import { applicationConfiguration } from './application.configuration'
+import AmazonS3URI from 'amazon-s3-uri'
 
 interface JobData {
   applicationId: string
   attachmentUrl: string
 }
 
+interface JobResult {
+  attachmentKey: string
+  resultUrl: string
+}
+
 @Processor('upload')
 export class UploadProcessor {
   constructor(
     private readonly applicationService: ApplicationService,
     private readonly fileStorageService: FileStorageService,
+    @Inject(applicationConfiguration.KEY)
+    private readonly config: ConfigType<typeof applicationConfiguration>,
   ) {}
 
   @Process('upload')
-  async handleUpload(job: Job) {
-    const { attachmentUrl }: JobData = job.data
-    console.log('Start uploading...')
-    const { bucket, key } = AmazonS3URI(attachmentUrl)
-    const destinationBucket = 'testing-islandis-copy'
-    const region = 'eu-west-1'
+  async handleUpload(job: Job<JobData>): Promise<JobResult> {
+    const { attachmentUrl, applicationId } = job.data
+    const destinationBucket = this.config.attachmentBucket
+
+    if (!destinationBucket) {
+      throw new Error('Application attachment bucket not configured.')
+    }
 
-    return await this.fileStorageService.copyObjectFromUploadBucket(
-      key,
+    const { key: sourceKey } = AmazonS3URI(attachmentUrl)
+    const destinationKey = `${applicationId}/${sourceKey}`
+    const url = await this.fileStorageService.copyObjectFromUploadBucket(
+      sourceKey,
       destinationBucket,
-      region,
-      bucket,
+      destinationKey,
     )
+
+    const resultUrl = `https://${destinationBucket}.s3-${this.config.region}.amazonaws.com/${destinationKey}`
```

Comment:
You already have the url from the `copyObjectFromUploadBucket`

Reply:
Good catch. I went back and forth on where the URLs should be managed.